### ISSUE TITLE
unusual_byte_groupings triggers inconsistently for byte-grouped binary literals

### ISSUE THREAD

**Issue by lucy** [rust-lang/rust-clippy#9183]
### Summary

See title. Personally I don't think grouping binary literals by byte is unusual.

### Lint Name

unusual_byte_groupings

### Reproducer

```rust
const TEST1: u16 = 0b11111111_11111111;
```
Outputs:
```
warning: digits of hex or binary literal not grouped by four
 --> src/term.rs:1:20
  |
1 | const TEST1: u16 = 0b11111111_11111111;
  |                    ^^^^^^^^^^^^^^^^^^^ help: consider: `0b1111_1111_1111_1111`
```

While this doesn't give any warnings:
```rust
const TEST2: u16 = 0b11111111_1111_1111;
```

### Version

```text
rustc 1.64.0-nightly (87588a2af 2022-07-13)
binary: rustc
commit-hash: 87588a2afd9ca903366f0deaf84d805f34469384
commit-date: 2022-07-13
host: x86_64-unknown-linux-gnu
release: 1.64.0-nightly
LLVM version: 14.0.6
```

**Comment by reitermarkus** [comment]
I group my `u16` `bitflags` by byte, i.e. `0b00000000_01000000`, so I also encountered this.

**Comment by llogiq** [comment]
Fixed by #10353
